test/functionalTest/cases/3068_non_legacy_notifications/non_legacy_notifications_with_accumulator.test

```diff
@@ -0,0 +1,113 @@
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Simple forwarded query in NGSIv2
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 38,186-187,235 IPV4
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Register entities of type T, with ID .*, and attribute A1, for the accumulator
+# 02. Query entities of type T - get empty response as the accumulator doesn't respond
+# 03. Dump the accumulator, see the forwarded query
+#
+
+echo "01. Register entities of type T, with ID .*, and attribute A1, for the accumulator"
+echo "=================================================================================="
+payload='{
+  "dataProvided": {
+    "entities": [
+      {
+        "type": "T",
+        "idPattern": ".*"
+      }
+    ],
+    "attrs": [ "A1" ]
+  },
+  "provider": {
+    "http": {
+      "url": "http://localhost:'${LISTENER_PORT}'/v2"
+    },
+    "legacyForwarding": false
+  }
+}'
+orionCurl --url /v2/registrations --payload "$payload"
+echo
+echo
+
+
+echo "02. Query entities of type T - get empty response as the accumulator doesn't respond"
+echo "===================================================================================="
+orionCurl --url /v2/entities?type=T
+echo
+echo
+
+
+echo "03. Dump the accumulator, see the forwarded query"
+echo "================================================="
+accumulatorDump
+echo
+echo
+
+
+--REGEXPECT--
+01. Register entities of type T, with ID .*, and attribute A1, for the accumulator
+==================================================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/registrations/REGEX([0-9a-f\-]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+02. Query entities of type T - get empty response as the accumulator doesn't respond
+====================================================================================
+HTTP/1.1 200 OK
+Content-Length: 2
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+[]
+
+
+03. Dump the accumulator, see the forwarded query
+=================================================
+GET http://localhost:REGEX(\d+)/v2/entities?type=T
+Fiware-Servicepath: /
```

Comment:
Registration use

```
 01. Register entities of type T, with ID .*, and attribute A1, for the accumulator
```

so I wonder why we are not seeing `attrs=A1` here... Maybe a but in the implementation?

Reply:
Seems like a commit is missing. Pretty sure I implemented that (not in the first run, but afterwards).
Checking ...

Reply:
(Travis is currently failing... maybe due the missing commit?)

Reply:
So, I looked into this test.
All seems just OK.
The query in step 2 is like this:
```
orionCurl --url /v2/entities?type=T
```
So, that is what is forwarded.

What you propose, _to check the registration data and extract the "smallest" part of the query and the registration_ is part of the final implementation for fully working forwarding.
For that we need a complete rewrite of the forwarding mechanism. Not worth the trouble to do it in this "attempt to make forwarding a little better".

Reply:
No need of a complete rewrite of the forwarding mechanism. As I explain at https://github.com/telefonicaid/fiware-orion/issues/3463 is just a matter of improving the logic in a couple of points. But that's another history to be addresses in another issue.

Reply:
(It seems I do a missplaced comment. In fact https://github.com/telefonicaid/fiware-orion/pull/3467#discussion_r271605574 actually applies here :)

Reply:
Fixed in 19efa15